src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx

```diff
@@ -192,6 +200,34 @@ const ContentMessageComponent: React.FC<ContentMessageProps> = ({
     }
   };
 
+  const mainViewModel = useMainViewModel();
+  const {content: contentViewModel} = mainViewModel;
+  const {messageRepository} = contentViewModel;
+
+  const handleRetrySending = async () => {
+    const firstAsset = message.getFirstAsset();
+
+    if (firstAsset instanceof Text) {
+      const messageId = message.id;
+      const messageText = message.getFirstAsset().text;
+      const mentions: MentionEntity[] = firstAsset.mentions() && firstAsset.mentions();
+      const incomingQuote = message.quote();
+
+      const isOutgoingQuote = (quoteEntity: QuoteEntity): quoteEntity is OutgoingQuote => {
+        return quoteEntity.hash !== undefined;
+      };
+
+      const quote: OutgoingQuote | undefined =
+        incomingQuote && isOutgoingQuote(incomingQuote) ? (incomingQuote as OutgoingQuote) : undefined;
+
+      await messageRepository.sendTextWithLinkPreview(conversation, messageText, mentions, quote, messageId);
+    }
+  };
+
+  const handleDiscard = async () => {
+    await messageRepository.deleteMessageById(conversation, message.id);
+  };
+
```

Comment:
This logic doesn't really belong there. 
The fact that you needed to inject the `messageRepository` in this component is an indicator that the logic might not be in the correct layer of the app. 

If you look at the codebase, you will see that the `MessageWrapper` already has access to the `MessageRepository`, so it should be to one implementing `retry` and `discard`. 

This component should have 2 callbacks as props `onRetry` and `onDiscard` and this logic should live in the `MessageWrapper`